File: DarkOrbit 10.0/Net/netty/commands/PetStatusCommand.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Ow.Utils;

namespace Ow.Net.netty.commands
{
    class PetStatusCommand
    {
        public const short ID = 22037;

        public static byte[] write(int petId, int petLevel, double petExperiencePoints, double petExperiencePointsUntilNextLevel, int petHitPoints, int petHitPointsMax, int petShieldEnergyNow, int petShieldEnergyMax, int petCurrentFuel, int petMaxFuel, int petSpeed, string petName)
        {
            var param1 = new ByteArray(ID);
            param1.writeInt(petHitPoints << 15 | petHitPoints >> 17);
            param1.writeInt(petMaxFuel << 5 | petMaxFuel >> 27);
            param1.writeInt(petId << 13 | petId >> 19);
            param1.writeInt(petLevel << 7 | petLevel >> 25);
            param1.writeInt(petShieldEnergyNow >> 3 | petShieldEnergyNow << 29);
            param1.writeDouble(petExperiencePointsUntilNextLevel);
            param1.writeInt(petShieldEnergyMax << 2 | petShieldEnergyMax >> 30);
            param1.writeUTF(petName);
            param1.writeInt(petSpeed >> 9 | petSpeed << 23);
            param1.writeInt(petCurrentFuel >> 11 | petCurrentFuel << 21);
            param1.writeDouble(petExperiencePoints);
            param1.writeInt(petHitPointsMax << 4 | petHitPointsMax >> 28);
            return param1.ToByteArray();
        }
    }
}

File: DarkOrbit 9.0/DarkOrbit 9.0 Emulator/src/simulator/netty/serverCommands/PetStatusCommand.java
package simulator.netty.serverCommands;

import java.io.DataOutputStream;
import java.io.IOException;
import simulator.netty.ServerCommand;

/**
Created by LEJYONER on 21/09/2017.
*/

public class PetStatusCommand
        implements ServerCommand {

    public static int ID = 20686;

    public int petId; 
    public int petLevel; 
    public double petExperiencePoints; 
    public double petExperiencePointsUntilNextLevel; 
    public int petHitPoints;
    public int petHitPointsMax; 
    public int petShieldEnergyNow; 
    public int petShieldEnergyMax; 
    public int petCurrentFuel; 
    public int petMaxFuel;
    public int petSpeed; 
    public String petName;

    
    public PetStatusCommand(int param1, int param2, double param3, double param4, int param5,
    		int param6, int param7, int param8, int param9, int param10, int param11, String param12) {
        this.petId = param1;
        this.petLevel = param2;
        this.petExperiencePoints = param3;
        this.petExperiencePointsUntilNextLevel = param4;
        this.petHitPoints = param5;
        this.petHitPointsMax = param6;
        this.petShieldEnergyNow = param7;
        this.petShieldEnergyMax = param8;
        this.petCurrentFuel = param9;
        this.petMaxFuel = param10;
        this.petSpeed = param11;
        this.petName = param12;
    }

    public void write(DataOutputStream param1) {
        try {
            param1.writeShort(ID);
            this.writeInternal(param1);
        } catch (IOException e) {
        }
    }

    protected void writeInternal(DataOutputStream param1) {
        try {
            param1.writeInt(this.petShieldEnergyNow >>> 8 | this.petShieldEnergyNow << 24);
            param1.writeInt(this.petId << 11 | this.petId >>> 21);
            param1.writeInt(this.petCurrentFuel >>> 4 | this.petCurrentFuel << 28);
            param1.writeDouble(this.petExperiencePointsUntilNextLevel);
            param1.writeUTF(this.petName);
            param1.writeDouble(this.petExperiencePoints);
            param1.writeInt(this.petHitPointsMax >>> 9 | this.petHitPointsMax << 23);
            param1.writeInt(this.petMaxFuel << 16 | this.petMaxFuel >>> 16);
            param1.writeInt(this.petShieldEnergyMax >>> 8 | this.petShieldEnergyMax << 24);
            param1.writeInt(this.petLevel << 13 | this.petLevel >>> 19);
            param1.writeInt(this.petSpeed >>> 15 | this.petSpeed << 17);
            param1.writeInt(this.petHitPoints >>> 5 | this.petHitPoints << 27);
            param1.writeShort(32638);
        } catch (IOException e) {
        }
    }

}
